_posts/2020-07-10-removing-negative-words

```diff
@@ -0,0 +1,86 @@
+---
+layout: post
+title: "Removing Negative Words From our Codebase"
+date: 2020-07-10 11:00:00
+reviewed: 2020-07-10 11:00:00
+categories: ["inclusion", "culture"]
+author: "abizzinotto"
+---
+
+The discussion around the use of problematic words / terms in technology isn't a new one. This issue [can be traced back to, at least, 2003](https://edition.cnn.com/2003/TECH/ptech/11/26/master.term.reut/) when a Los Angeles county worker filed a complaint with the city's Office of Affirmative Action Compliance after seeing "master" and "slave" in computer equipment labels.
+
+This issue has resurfaced with recent events and, at OmbuLabs, we believe it is important to take it seriously. It is past time to remove negative words from our codebase. We have started taking the necessary steps to rename "master" branches to "main", as well as (and perhaps more importantly) remove any reference to the term "slave" and replace "whitelist / blacklist" with "allowlist / blocklist". Our team is actively working on that.
+
+But we understand that is not enough. This is not an action that effects change in its own. It's part of a larger commitment. In this article, I'll explain the context and motivation for this change and how it fits with our core values and log term vision as a company.
+
+<!--more-->
+
+## Context
+
+As far as the sofware community is concerned, efforts to move away from such terms have been underway for quite some time now. In 2014, [Drupal](https://www.drupal.org/) opted for the use of "primary / replica" instead of "master / slave" and, in the same year, [Django] (https://www.djangoproject.com/) opted to use "leader / follower". Then in 2018 the decision was made to remove the terminology from Python, same year that the IETF (Internet Engineering Task Force) published their [Terminology, Power and Oppressive Language](https://tools.ietf.org/id/draft-knodel-terminology-00.html#rfc.section.1.1) memo arguing for the terminology change and offering alternatives to be use instead of the offensive terms.
```

Comment:
I think there is a big confusion on the software terminology for `master/slave` in this post.

I'm %100 in favor of removing the `master/slave` terminology in system design and implementation, since is clearly racist because it implies one machine has control over the other. However, the use of the `master` terminology in a branching model is as racists as someone telling me it has a master degree or someone telling me it is a naval master as the rank for a naval officer, the context is not the same as a system having a `master/slave` relationship at all.

My problem is when we start to satanize things/words for the lack of understanding from where they come, renaming the branch from `master` to `main` is absolutely pointless and it just reflects our lack of understanding of where the `master/slave` terminology comes from the software perspective and just wanting to fit in the current GitHub CEO trend bandwagon.

Wikipedia even separates both terminologies.
<img width="1071" alt="Screen Shot 2020-07-02 at 8 02 29 PM" src="https://user-images.githubusercontent.com/4595174/86424165-ff29b700-bc9e-11ea-9985-d3bf0376c626.png">

Python, Django, Go and ZFZ
<img width="289" alt="Screen Shot 2020-07-02 at 6 21 30 PM" src="https://user-images.githubusercontent.com/4595174/86420689-9fc6a980-bc94-11ea-8122-134ff7d7382e.png">
<img width="304" alt="Screen Shot 2020-07-02 at 6 21 54 PM" src="https://user-images.githubusercontent.com/4595174/86420696-a3f2c700-bc94-11ea-821e-8d0a949b4ded.png">
<img width="277" alt="Screen Shot 2020-07-02 at 6 24 51 PM" src="https://user-images.githubusercontent.com/4595174/86420705-a9501180-bc94-11ea-8cb5-1fc6d8b39bcb.png">
<img width="391" alt="Screen Shot 2020-07-02 at 6 46 42 PM" src="https://user-images.githubusercontent.com/4595174/86420710-abb26b80-bc94-11ea-83b6-ed96f5366cf5.png">

You might wonder why all the projects that are mentioned in this post are still using `master` as a branch name even when they replaced their terminology from `master/slave` in their systems and that is because it has nothing to do with the `master/slave` terminology we want to abolish.

In my 10 years of programming I have never ever ever have thought of the master branch as in `master/slave` 
as the same way I would never ever associate someone telling me they have a master degree as racist, the slavery association simply doesn't exists.

There are not good or bad words, all depends on the context and given that we lack the racist context in the master branch terminology I don't see how we are achieving anything by changing it...

Reply:
@bronzdoc while I understand the question here and I think it is a very valid point that needs to be raised (and is actively being discussed elsewhere) I am not sure comments in the PR are the best way to do it (I don't want to write a bunch of stuff here for you to read) so I think this would be more beneficial to discuss on a call. What do you think?